src/Xamarin.Android.Build.Tasks/Tasks/FilterAssemblies.cs

```diff
@@ -40,6 +45,19 @@ public override bool Execute ()
 					var targetFrameworkIdentifier = assemblyDefinition.GetTargetFrameworkIdentifier (reader);
 					if (targetFrameworkIdentifier == TargetFrameworkIdentifier) {
 						output.Add (assemblyItem);
+						continue;
+					}
+					// Fallback to looking at references
+					if (string.IsNullOrEmpty (targetFrameworkIdentifier) && !string.IsNullOrEmpty (FallbackReference)) {
+						Log.LogDebugMessage ($"Checking references for: {assemblyItem.ItemSpec}");
+						foreach (var handle in reader.AssemblyReferences) {
```

Comment:
Does this need to be recursive?  As-is, this will only handle direct assembly references.

For example, `Xamarin.Forms.Core.dll` references `Xamarin.Forms.Platform.dll` -- which exists for multiple platforms -- and the `MonoAndroid10/Xamarin.Forms.Platform.dll` references `Xamarin.Forms.Platform.Android.dll` which references `Mono.Android.dll`.  (Deep breath.)

Thus, should we *effectively* emit this, i.e. "recursively" resolve references?

```xml
<ItemGroup>
  <_ResolvedUserAssemblies Include="Xamarin.Forms">
    <TargetFrameworkIdentifier>MonoAndroid</TargetFrameworkIdentifier>
</ItemGroup>
```

...though then *everything* could then be treated as MonoAndroid profile assemblies, which feels like it would defeat the purpose.

Thus, how thorough do we need to be?

Reply:
I have only found this one case where this `[assembly:TargetFramework]` was missing... It was [shipped to NuGet in 2015](https://www.nuget.org/packages/Refractored.Controls.CircleImageView/).

I built it from source: https://github.com/jamesmontemagno/CircleImageView-Xamarin.Android

And the assembly produced with VS 2019 has:
```csharp
[assembly: TargetFramework("MonoAndroid,Version=v5.0", FrameworkDisplayName = "Xamarin.Android v5.0 Support")]
```

Reply:
I’m overthinking this; Xamarin.Forms does *not* need to have `%(TargetFrameworkIdentifier)`=MonoAndroid.

Reason being that while, indirectly, mythical type `A` using Xamarin.Forms has a reference graph which eventually includes `Mono.Android.dll` (via `Xamarin.Forms.Platform.Android.dll`), `A` cannot use any types from `Mono.Android.dll` — even indirect inheritance! — without referencing it explicitly.

Reply:
I updated the commit message around this conversation 👍